src/luarocks/build.lua

```diff
@@ -94,10 +92,12 @@ local function check_macosx_deployment_target(rockspec)
    end
    if cfg.is_platform("macosx") and rockspec:format_is_at_least("3.0") and target then
       local version = util.popen_read("sw_vers -productVersion")
-      local versionminor = minor(version)
-      local targetminor = minor(target)
-      if targetminor > versionminor then
-         return nil, ("This rock requires Mac OSX 10.%d, and you are running 10.%d."):format(targetminor, versionminor)
+      if version:match("^%d+%.%d+%.%d+$") or version:match("^%d+%.%d+$") then
+         version = vers.parse_version(version)
+         local targetversion = vers.parse_version(target)
+         if targetversion > version then
+            return nil, ("This rock requires Mac OSX %s, and you are running %s."):format(targetversion, version)
```

Comment:
you might need `tostring` here; lua 5.1 doesn't call it for us on string.format arguments.

Reply:
You're right!